android/src/main/cpp/NativeProxy.cpp

```diff
@@ -357,10 +358,26 @@ void NativeProxy::installJSIBindings(
         jsi::Object yogaValues(rt);
         for (const auto &entry : *values) {
           try {
-            auto key =
-                jsi::String::createFromAscii(rt, entry.first->toStdString());
-            auto value = stod(entry.second->toStdString());
-            yogaValues.setProperty(rt, key, value);
+            std::string keyString = entry.first->toStdString();
+            std::string valueString = entry.second->toStdString();
+            auto key = jsi::String::createFromAscii(rt, keyString);
+            if (keyString == "currentTransformMatrix" ||
+                keyString == "targetTransformMatrix") {
+              std::vector<float> transformMatrixList;
+              std::istringstream stringStream(valueString);
+              std::copy(
+                  std::istream_iterator<float>(stringStream),
+                  std::istream_iterator<float>(),
+                  std::back_inserter(transformMatrixList));
+              jsi::Array matrix(rt, 9);
+              for (int i = 0; i < 9; i++) {
+                matrix.setValueAtIndex(rt, i, transformMatrixList[i]);
```

Comment:
Can you add a length assertion on `transformMatrixList` here, so we don't read from after the end if we pass a wrong value?